.gitignore

```diff
@@ -3,7 +3,6 @@
 .DS_Store
 
 /beta
-/build
```

Comment:
Our Readme documents:
```
cmake -B build -DCMAKE_BUILD_TYPE=RelWithDebInfo
```

So we should keep that `/build` folder ignored: there is no reason to version anything from it.

Reply:
the `build` dir is ignored, see the second commit. It will write `build/.gitignore` with the contents being `*`

Reply:
I see. Although if one git pull to your commit, we'll be in an intermediate state: we got a `build` folder from a previous older build but we don't have that new .gitignore yet, so all our past build files appear in the git list.
It might be nicer to keep `/build` for a transitioning period.

Reply:
It checks every time cmake runs if that files exists, so technically it would be
```sh
git pull
cmake --build build
```
There is the small period between when you pull the cmake changes and reconfigure your build that `build` will show as a dirty, but then once cmake is run that will go away.

Reply:
It is almost always a misconfiguration to build your project in-tree. What you're describing is exactly why it's a bad idea, too. If you're working with multiple builds, you are constantly overwriting build artifacts in the tree and confusing yourself. Obviously, a developer would not do that and would use a separate build dir per configuration. 

For regular users and downstream packagers, I see no reason why they need this:
- A regular user could accidentally mix/match build types and get strange, and hard-to-debug errors
- A packager never needs to build in-tree, and in fact I see no package where this is done
- Transmission's own documentation says to use `-B build`. (And if there are places it doesn't, I'll update that as part of this PR)


Furthermore, your second usecase:
>I have multiple build dirs with cmake, like one for Release (build-Rel), one for Debug (just build), and one for Xcode generator (-G Xcode) tests (build-Xcode).
> I would really be nice to have /build* ignore.

Is handled by this PR, look at these lines: https://github.com/transmission/transmission/blob/ff115173e23869a679ce67a45a46ffb990e71fbe/CMakeLists.txt#L29-L33

**Any** and **all** `CMAKE_BINARY_DIR` (i.e. the dir set with `-B`) will have a `.gitignore` file placed into it, if it doesn't already exist. So in fact `build-XCode`, `build-Rel` and `build` will all be ignored implicitly. You never have to think about what your build dir is called.